prowler/providers/aws/services/ec2/ec2_service.py

```diff
@@ -407,6 +411,46 @@ def __get_ebs_encryption_settings__(self, regional_client):
                 f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
 
+    def __get_snapshot_block_public_access_state__(self, regional_client):
+        try:
+            snapshot_in_region = False
+            for snapshot in self.snapshots:
```

Comment:
You need to add another `try/except` clause within the loop to handle exceptions there and not to exit the loop execution.